Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs

```diff
@@ -0,0 +1,244 @@
+        public float UpdatePeriod_DuringScanning = 1.0f;
+        [Tooltip("Update period used after the scanning process is completed")]
+        public float UpdatePeriod_AfterScanning = 4.0f;
+
+        // Properties
+        /// <summary>
+        /// Switch used by the entire SpatialUnderstanding module to activate processing.
+        /// </summary>
+        public bool AllowSpatialUnderstanding
+        {
+            get
+            {
+#if UNITY_METRO && !UNITY_EDITOR
+                return true;
+#endif
+                return false;
+            }
+        }
+        /// <summary>
+        /// Reference to the SpatialUnderstandingDLL class (wraps the understanding dll functions).
+        /// </summary>
+        public SpatialUnderstandingDll UnderstandingDLL { get; private set; }
+        /// <summary>
+        /// Reference to the SpatialUnderstandingSourceMesh behavior (input mesh data for the understanding module).
+        /// </summary>
+        public SpatialUnderstandingSourceMesh UnderstandingSourceMesh { get; private set; }
+        /// <summary>
+        /// Reference to the UnderstandingCustomMesh behavior (output mesh data from the understanding module).
+        /// </summary>
+        public SpatialUnderstandingCustomMesh UnderstandingCustomMesh { get; private set; }
+        /// <summary>
+        /// Indicates the current state of the scan process
+        /// </summary>
+        public ScanStates ScanState
+        {
+            get
+            {
+                return scanState;
+            }
+            private set
+            {
+                scanState = value;
+                if (ScanStateChanged != null)
+                {
+                    ScanStateChanged();
+                }
+
+                // Update scan period, based on state
+                SpatialMappingManager.Instance.GetComponent<SpatialMappingObserver>().TimeBetweenUpdates = (scanState == ScanStates.Done) ? UpdatePeriod_AfterScanning : UpdatePeriod_DuringScanning;
+            }
+        }
+        /// <summary>
+        /// Indicates the the scanning statistics are still being processed.
+        /// Request finish should not be called when this is true. 
+        /// </summary>
+        public bool ScanStatsReportStillWorking
+        {
+            get
+            {
+                if (SpatialUnderstanding.Instance.AllowSpatialUnderstanding)
+                {
+                    SpatialUnderstandingDll.Imports.PlayspaceStats stats = SpatialUnderstanding.Instance.UnderstandingDLL.GetStaticPlayspaceStats();
+                    return (stats.IsWorkingOnStats != 0);
+                }
+                return false;
+            }
+        }
+
+        // Events
+        /// <summary>
+        /// Event indicating that the scan state has changed
+        /// </summary>
+        public event Action ScanStateChanged;
+
+        // Privates
+        private ScanStates scanState;
+
```

Comment:
This one's a NIT for sure, but member variables, event definitions and properties are interspersed throughout the class. I would like to see them better grouped together. I'd especially like to see all member variables defined at the top.